[Feature] Accounts Page Pagination.

gertie-sheshe opened the pull request:
## Description

1. Account creation in List.
2. Account deletion in List.
3. Accounts pagination.

## Type of change

Please delete options that are not relevant.

- [ ] Chore (non-breaking change which does not add visible functionality but improves code quality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

kilemensi commented:
Merge conflicts @gertie-sheshe

KhadijaMahanga commented on src/components/AccountsList/index.js:
```diff
@@ -24,30 +26,53 @@ function AccountsList({
-  const fetcher = (url) => fetchJson(url);
-  const { data, mutate } = useSWR(apiUrl, fetcher);
+  const [existingAccounts, setExistingAccounts] = useState(accounts);
+  const [page, setPage] = useState();
```
@gertie-sheshe this shouldn't be undefined.. or your initial call to fetch data will fail. So either you pass a props or use default of 1 (i.e first page)
```suggestion
  const [page, setPage] = useState(pageProp || 1);
```

KhadijaMahanga commented on src/components/AccountsList/index.js:
```diff
@@ -24,30 +26,53 @@ function AccountsList({
+    return fetchJson(listURL);
+  };
+
+  const { data, mutate } = useSWR(
+    [`${apiUrl}/?accounts=true`, page, pageSize],
+    fetcher
+  );
+
+  const { data: accountsData, mutate: mutateAccountsData } = useSWR(
+    `${apiUrl}/?accounts=true`,
+    fetcher
+  );
```
@gertie-sheshe I didn't know you call `useSWR` hook twice. 

Anyway, isn't this one and the same?  Both using the same url `${apiUrl}/?accounts=true`

gertie-sheshe replied:
Yes you can. Did it because we need all accounts during addition of a new one. @KhadijaMahanga 

No they aren't the same, the first one has the url plus arguments to pass to the fetcher function i.e [`${apiUrl}/?accounts=true`, page, pageSize]

KhadijaMahanga replied:
@gertie-sheshe if we are using it during creation, then I don't think it should be in a hook. It should be called during creation only

gertie-sheshe replied:
Fixed

KhadijaMahanga reviewed (commented):
@gertie-sheshe 
1. Delete account is disabled when I am on page 2 and there's a single item. I think we should disable it only when we are on page==1 and there's a single item.
2. Once I delete accounts, pagination doesn't update. See screenshot below => I had 5 accounts (2 pages), I deleted 2 accounts. But page 2 is still there and empty

https://user-images.githubusercontent.com/7962097/152139788-d1556c67-e6b7-4641-a2ed-a3166c31615f.mov

3. I don't understand why we have `existingAccounts` variable

gertie-sheshe commented:
> @gertie-sheshe
> 
> 1. Delete account is disabled when I am on page 2 and there's a single item. I think we should disable it only when we are on page==1 and there's a single item.
> 2. Once I delete accounts, pagination doesn't update. See screenshot below => I had 5 accounts (2 pages), I deleted 2 accounts. But page 2 is still there and empty
> 
>  Screen.Recording.2022-02-02.at.13.39.53.mov 
> 3. I don't understand why we have `existingAccounts` variable

Fixed all these

KhadijaMahanga reviewed (approved):
🚀
